### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by shcrela**
added the script "wdfReader.py" which enables us to read the binary .wdf files and also added the "combine_PCA_WDF.py" which combines the wdfREader with deconvolution script, as well as providig some supplementary visualizations (heatmap of mixing coeffs and some other stuff)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Data/
```
Comment (emmanuelle) [inline .gitignore:1]: Maybe it'd be good to add a Data folder to the version control at some point. Not all data though (a few Mo max).

File: .ipynb_checkpoints/Untitled-checkpoint.ipynb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (emmanuelle) [inline .ipynb_checkpoints/Untitled-checkpoint.ipynb:1]: Does this file belong to the version control?

File: .gitignore (hunk 1 above)
Comment (emmanuelle) [reply .gitignore:1]: Please also add .pyc files to the gitignore

File: .ipynb_checkpoints/wdfReader-checkpoint.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
+from __future__ import print_function
```
Comment (emmanuelle) [inline .ipynb_checkpoints/wdfReader-checkpoint.py:1]: is this file needed in the version control?

File: combine_PCA_WDF.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    
+'''
+
+#filename = 'Data/Test-Na-SiO2 0079 -532nm-obj100-p100-10s over night carto.wdf'#scan_type 1, measurement_type 3
+#filename = 'Data/Test-Na-SiO2 0079 droplet on quartz -532nm-obj50-p50-15s over night_Copy_Copy.wdf'#scan_type 2, measurement_type 2
+#filename = 'Data/Test quartz substrate -532nm-obj100-p100-10s.wdf'#scan_type 2, measurement_type 1
+#filename = 'Data/Hamza-Na-SiO2-532nm-obj100-p100-10s-extended-cartography - 1 accumulations.wdf'#scan_type 2 (wtf?), measurement_type 3
+#filename = 'Data/M1SCMap_2_MJ_Truncated_CR2_NF50_PCA3_Clean2_.wdf'
+#filename = 'Data/M1ANMap_Depth_2mm_.wdf'
+filename = 'Data/M1SCMap_depth_.wdf'
```
Comment (emmanuelle) [inline combine_PCA_WDF.py:48]: is this file too large to be added to the repo?

File: combine_PCA_WDF.py [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+snake = True # The scanning mode: either always from left to right (snake = False), either left->right right->left (snake = True)
+n_components=5
+# one should always check if the spectra were recorded with the dead pixels included or not
+# It turns out that firs 10 and the last 16 pixels on the Renishaw SVI spectrometer detector are reserved, 
+# and no signal is recorded on those pixels by the detector. So we should either enter these parameters inside the Wire settings
+# or if it's not done, remove those pixels here manually
+# Furthermore, we sometimes want to perform the deconvolution only on a part of the spectra, so here you define the part that interests you
+ssss = slice(575,950)
```
Comment (emmanuelle) [inline combine_PCA_WDF.py:56]: Please make sure that all variable names are as informative as possible. E.g _slice, spectra, reader... (a lot are already fine like laser_wavelength etc.)

File: combine_PCA_WDF.py [hunk 6]
```diff
@@ -0,0 +1,185 @@
+# -*- coding: utf-8 -*-
+import wdfReader
+import deconvolution
+import matplotlib.pyplot as plt
+from sklearn import decomposition
+import h5py
+import numpy as np
+import seaborn as sns; sns.set()
+from copy import copy
```
Comment (emmanuelle) [inline combine_PCA_WDF.py:9]: For numpy arrays, it's better to use np.copy

**Review by emmanuelle — COMMENTED** [review]
Thank for the PR! i have read throught the script to process maps, I still need to go to the module to read such files.

File: .ipynb_checkpoints/wdfReader-checkpoint.py (hunk 3 above)
Comment (shcrela) [reply .ipynb_checkpoints/wdfReader-checkpoint.py:1]: same as above

File: combine_PCA_WDF.py (hunk 4 above)
Comment (shcrela) [reply combine_PCA_WDF.py:48]: it's 82Mb

File: wdfReader.py [hunk 7]
```diff
@@ -0,0 +1,285 @@
+from __future__ import print_function
+import struct
+import numpy
+
+l_int16 = 2
```
Comment (emmanuelle) [inline wdfReader.py:5]: Here you could use the numpy data types and their attributes (itemsize, etc.) for more readability. See https://docs.scipy.org/doc/numpy-1.15.1/reference/arrays.dtypes.html
